Occasional test failure

### Describe the issue

`weblate/trans/tests/test_models.py::ProjectTest::test_add_suggestion_validation` sometimes fails:

```
__________________ ProjectTest.test_add_suggestion_validation __________________

self = <weblate.trans.tests.test_models.ProjectTest testMethod=test_add_suggestion_validation>

    def test_add_suggestion_validation(self) -> None:
        with transaction.atomic():
            component = self.create_po(
                suggestion_voting=True, suggestion_autoaccept=True
            )
            user = create_test_user()
            another_user = create_another_user()
            translation = component.translation_set.get(language_code="cs")
            unit: Unit = translation.unit_set.all()[0]
    
            unit.translate(user, "Translation of unit ", STATE_TRANSLATED)
    
            # check that another user submitting the same target raises an error
>           with self.assertRaises(SuggestionSimilarToTranslationError):
E           AssertionError: SuggestionSimilarToTranslationError not raised

weblate/trans/tests/test_models.py:164: AssertionError
=============================== warnings summary ===============================
```

This happens very rarely.

### I already tried

- [x] I've read and searched [the documentation](https://docs.weblate.org/).
- [x] I've searched for similar filed issues in this repository.

### Steps to reproduce the behavior

Not reproducible, example log: https://github.com/WeblateOrg/weblate/actions/runs/13372271391/job/37343355988

### Expected behavior

_No response_

### Screenshots

_No response_

### Exception traceback

```pytb

```

### How do you run Weblate?

_No response_

### Weblate versions

_No response_

### Weblate deploy checks

```shell

```

### Additional context

_No response_

--- a/weblate/trans/tests/test_models.py
+++ b/weblate/trans/tests/test_models.py
@@ -144,7 +144,7 @@ class ProjectTest(RepoTestCase):
             )
             user = create_test_user()
             translation = component.translation_set.get(language_code="cs")
-            unit = translation.unit_set.all()[0]
+            unit = translation.unit_set.get(source="Hello, world!\n")
             suggestion = Suggestion.objects.add(unit, ["Test"], None)
             Vote.objects.create(suggestion=suggestion, value=Vote.POSITIVE, user=user)
         component.project.delete()
@@ -157,7 +157,7 @@ class ProjectTest(RepoTestCase):
             user = create_test_user()
             another_user = create_another_user()
             translation = component.translation_set.get(language_code="cs")
-            unit: Unit = translation.unit_set.all()[0]
+            unit = translation.unit_set.get(source="Hello, world!\n")
 
             unit.translate(user, "Translation of unit ", STATE_TRANSLATED)
 

